src/status_im/extensions/views.cljs

```diff
@@ -0,0 +1,68 @@
+(ns status-im.extensions.views
+  (:require-macros [status-im.utils.views :refer [defview letsubs]])
+  (:require [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.ui.components.camera :as camera]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.status-bar.view :as status-bar]
+            [status-im.ui.components.toolbar.view :as toolbar]
+            [status-im.ui.components.toolbar.actions :as actions]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.i18n :as i18n]
+            [status-im.ui.screens.profile.photo-capture.styles :as styles]
+            [status-im.utils.image-processing :as image-processing]))
+
+;; ensure photo taken or picked are not too big
+(def max-width 1024)
+(def max-height 1024)
+
+(defn- process-image-and-finish [path context]
+  (let [on-success (fn [base64]
+                     (re-frame/dispatch [:extensions/camera-picture-taken base64 context]))
+        on-error   (fn [type error]
+                     (re-frame/dispatch [:extensions/camera-error error context]))]
+    (image-processing/img->base64 path on-success on-error max-width max-height)))
+
+(defview take-picture []
+  (letsubs [context [:get-screen-params]
+            camera-ref (reagent/atom false)]
+    [react/view styles/container
+     [status-bar/status-bar]
+     [toolbar/toolbar nil
+      [toolbar/nav-button (actions/back
+                           #(do
+                              (re-frame/dispatch [:extensions/camera-cancel context])
+                              (re-frame/dispatch [:navigate-back])))]
+      [toolbar/content-title "send picture"]]
+     [camera/camera {:style         {:flex 1}
+                     :aspect        (:fill camera/aspects)
+                 ;    :captureQuality "480p"
+                     :captureTarget (:disk camera/capture-targets)
+                 ;    :type          "front"
+                     :ref           #(reset! camera-ref %)}]
+     [react/view styles/button-container
+      [react/view styles/button
+       [react/touchable-highlight {:on-press (fn []
+                                               (let [camera @camera-ref]
+                                                 (-> (.capture camera)
+                                                     (.then #(process-image-and-finish (.-path %) context))
+                                                     (.catch #(re-frame/dispatch [:extensions/camera-error % context])))))}
+        [react/view
+         [icons/icon :icons/camera {:color :white}]]]]]]))
+
+(defn- open-image-picker [context]
+  (react/show-image-picker
+   (fn [image]
+     (let [path (get (js->clj image) "path")]
+       (process-image-and-finish path context)))))
+
+(defn pick-or-take-picture-list-selection [context]
+  {:title   "send picture"
```

Comment:
i18n/label

Reply:
@flexsurfer do I just have to add the english translation to **translations/en.json**?

Reply:
timeout: yes assumed :)